features/desktop/unhandled_errors.feature

```diff
@@ -13,6 +13,50 @@ Feature: Reporting unhandled events
             | Main.RunScenario(System.String scenario)         | |
             | Main.Start()               | |
 
+    @windows_only
+    Scenario: Windows device and app data
+        When I run the game in the "UncaughtException" state
+        And I wait to receive an error
+        Then the error is valid for the error reporting API sent by the Unity notifier
+        And the exception "errorClass" equals "ExecutionEngineException"
+        And the exception "message" equals "Promise Rejection"
+        And the event "unhandled" is false
+        And custom metadata is included in the event
+        And the first significant stack frame methods and files should match:
+            | Main.DoUnhandledException(Int64 counter) | Main.DoUnhandledException(System.Int64 counter) |
+            | Main.RunScenario(System.String scenario)         | |
+            | Main.Start()               | |
+        
+        #device metadata
+        And the event "device.charging" is not null
+        And the event "device.batteryLevel" is not null
+        And the event "device.freeDisk" is not null
+        And the event "device.freeMemory" is not null
+        And the event "device.hostname" is not null
+        And the event "device.id" is not null
+        And the event "device.locale" is not null
+        And the event "device.manufacturer" is not null
+        And the event "device.model" is not null
+        And the event "device.osName" is not null
+        And the event "device.osVersion" is not null
+        And the event "device.runtimeVersions" is not null
+        And the event "device.screenDensity" is not null
+        And the event "device.screenResolution" is not null
+        And the event "device.time" is not null
```

Comment:
Would it be possible to validate that this is a timestamp, and that other app/device fields are non-zero integers etc?

Reply:
I wrote this following the example of other metadata tests. @twometresteve are there better ways i could be asserting here?

Reply:
We do have an existing step that the Android (and probably others) use:
```    
And the event "device.time" is a timestamp
```

This Cucumber step is defined in MazeRunner and I can show you the docs that we have for them (they're not perfect, but could well help).

For Fractal's other suggestions, I'm not sure that we already have steps defined - but we can easily add them.  Let's talk IRL about that!

Reply:
Added timestamp and non 0 int checks